django/db/models/expressions.py

```diff
@@ -375,7 +375,19 @@ def __eq__(self, other):
 
     def __hash__(self):
         path, args, kwargs = self.deconstruct()
-        return hash((path,) + args + tuple(kwargs.items()))
+        return hash(_ensure_hashable((path, args, kwargs)))
+
+
+def _ensure_hashable(value):
```

Comment:
I don't know if this is the best solution. It seems to add overhead (extra iteration) for a problem that's only present on PostgreSQL.

Reply:
It will be used whenever somebody tries to group by an expression that includes a list or dict, which currently is PostgreSQL but could be others in future.

Another option would be to implement `BaseExpression.__hash__` as:

```python
try: return hash(self.deconstruct())
except TypeError: return 0  # effectively fall back to using __eq__
```